raphtory/src/db/view_api/time.rs

```diff
@@ -115,16 +115,45 @@ impl<T: TimeOps + Clone + 'static> WindowSet<T> {
             }
     }
 
-    /// Returns the time index of this window set
-    pub fn time_index(&self, center: bool) -> Box<dyn Iterator<Item = i64> + Send> {
-        if center {
-            Box::new(self.clone().map(|view| {
-                view.start().unwrap() + ((view.end().unwrap() - view.start().unwrap()) / 2)
-            }))
-        } else {
-            Box::new(self.clone().map(|view| view.end().unwrap() - 1))
+    /// Returns the centered time index of this window set
+    pub fn center_time_index(&self) -> CentredTimeIndex<T> {
+        CentredTimeIndex {
+            windowset: self.clone(),
         }
     }
+
+    /// Returns the end time index of this window set
+    pub fn end_time_index(&self) -> EndTimeIndex<T> {
```

Comment:
Instead of having CentredTimeIndex and EndTimeIndex, could we have only one TimeIndex struct with the center attribute inside, so we check that flag in the implementation of `next`? That way we can maintain the `time_index()` function as it was instead of having two versions of it, which maybe is a little confusing for users. Actually that simplifies the python implementation for `time_index`, that can just rely on this one instead.